account_sdk/src/deploy_contract/deployment.rs

```diff
@@ -0,0 +1,69 @@
+use starknet::{
+    accounts::SingleOwnerAccount,
+    contract::ContractFactory,
+    core::types::{FieldElement, InvokeTransactionResult},
+    providers::{JsonRpcClient, Provider},
+    signers::Signer,
+};
+
+use crate::suppliers::{PredeployedClientSupplier, RpcClientSupplier};
+
+use super::pending::PendingDeployment;
+
+pub struct CustomAccountDeployment<'a, Q> {
+    supplier: &'a Q,
```

Comment:
what is supplier? can't we use JsonRpcClient? We don't need to support gateway. Perhaps starknet-rs needs to be updated? It recently remove the Provider interface in favor of a concrete JsonRpc type

Reply:
The idea behind the supplier was to supply consts for different networks in an abstract way (no consts).
I've recently noticed that the UDC addresses and FeeToken addresses are the same for different networks (initially I was thinking that they are specific to the network) so for now the supplier's role is to provide prefunded accounts details.
Thanks to such implementation we are are able to easily switch the tests between `katana` and `devnet` by simply reusing the same test and changing `KatanaSupplier` into `DevnetSupplier`.
In my opinion such abstraction may be useful when we start interacting with StarkNet mainnet.

I suggest we use JsonRpcClient as you've suggested and we will move the supplier to the tests for now, because it's the only place it's used for now but I would prefer not to delete it completely as it's already developed.

Reply:
We should be able to have compatibility across them without the Supplier, since we can assume those contracts are always constant across networks. Let's remove it, it's better to have less code